config/config.go

```
@@ -0,0 +1,103 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/user"
+	"runtime"
+)
+
+func Load() Config {
+
+	// Checks User's OS, then looks for the config file and tries to read the
+	// data.
+	// If there is no config file, creates a deafult one.
+	// Returns Config struct that can be used in main package.
+
+	currentOs := runtime.GOOS
+
+	var configPath string
+	switch currentOs {
+	case "windows":
+		configPath = "./config.ini"
+	default:
+		configPath = "./config.ini" // placeholder?
+	}
+
+	conf := Config{}
+	if exists(configPath) {
+		fmt.Println("Found existing config file.")
+		conf = loadExistingConfig(configPath)
+		// should check if everything is loaded successfully, but I dunno
+		// how.
+	} else {
+		fmt.Println("Config file not found. Creating a default one.")
+		conf = createDefaultConfig(configPath)
+	}
+	return conf
+}
+
+func exists(filepath string) bool {
+	//this just returns true or false if file already exists at the provided
+	//path.
+
+	info, err := os.Stat(filepath)
+	if os.IsNotExist(err) {
+		return false
+	} else {
+		return !info.IsDir()
+	}
+}
+
+type Config struct {
+	// an object that gets returned to the main package, containing all the
+	// needed parameters as fields.
+	Name string
+	Os   string
+}
+
+func loadExistingConfig(filepath string) Config {
+	// loads data from json at the provided filepath, creates a config struct
+	// with it
+	conf := Config{}
+
+	data, err := os.ReadFile(filepath)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	json.Unmarshal(data, &conf)
```

Don't skip error checks :)  
Could be a good chance to set up your code editor so that it underlines ignored errors in lines like this.

---

Okay this might be a silly question but I'm not sure how to catch the error here, syntactically.

Do I just write:
`err = json.Unmarshal(data, &conf)`
and then catch it as usual?

---

Exactly! You can even do it like this:
```
if err := json.Unmarshal(data, &conf); err != nil {
  // error handling goes here
}
```

In this case `err` will only live in the scope of the `if` block. Usually this approach doesn't work for cases when you have multiple returns, since the `result` variable in `if result, err := someFunc(); err != nil {}` also won't exist outside of this err block, and you probably want to use it after the error check as well.
But here this function only returns `error`, so it's perfectly fine to do it.